patches/api/0180-Mob-Spawner-API-Enhancements.patch

```diff
@@ -37,5 +211,206 @@ index cb447a4ad5a9dce7c98999a5d7fcd6111fc9b10e..5bbae759ce39d42886994e500fd9454e
+-     * <br>
+-     * Default value is 4.
+-     *
+-     * @return the current spawn count
+-     */
+-    public int getSpawnCount();
+-
+-    /**
+-     * Set how many mobs attempt to spawn.
+-     *
+-     * @param spawnCount the new spawn count
+-     */
+-    public void setSpawnCount(int spawnCount);
+-
+-    /**
+-     * Set the new maximum amount of similar entities that are allowed to be
+-     * within spawning range of this spawner.
+-     * <br>
+-     * If more than the maximum number of entities are within range, the spawner
+-     * will not spawn and try again with a new {@link #getDelay()}.
+-     * <br>
+-     * Default value is 16.
+-     *
+-     * @return the maximum number of nearby, similar, entities
+-     */
+-    public int getMaxNearbyEntities();
+-
+-    /**
+-     * Set the maximum number of similar entities that are allowed to be within
+-     * spawning range of this spawner.
+-     * <br>
+-     * Similar entities are entities that are of the same {@link EntityType}
+-     *
+-     * @param maxNearbyEntities the maximum number of nearby, similar, entities
+-     */
+-    public void setMaxNearbyEntities(int maxNearbyEntities);
+-
+-    /**
+-     * Get the maximum distance(squared) a player can be in order for this
+-     * spawner to be active.
+-     * <br>
+-     * If this value is less than or equal to 0, this spawner is always active
+-     * (given that there are players online).
+-     * <br>
+-     * Default value is 16.
+-     *
+-     * @return the maximum distance(squared) a player can be in order for this
+-     * spawner to be active.
+-     */
+-    public int getRequiredPlayerRange();
+-
+-    /**
+-     * Set the maximum distance (squared) a player can be in order for this
+-     * spawner to be active.
+-     * <br>
+-     * Setting this value to less than or equal to 0 will make this spawner
+-     * always active (given that there are players online).
+-     *
+-     * @param requiredPlayerRange the maximum distance (squared) a player can be
+-     * in order for this spawner to be active.
+-     */
+-    public void setRequiredPlayerRange(int requiredPlayerRange);
+-
+-    /**
+-     * Get the radius around which the spawner will attempt to spawn mobs in.
+-     * <br>
+-     * This area is square, includes the block the spawner is in, and is
+-     * centered on the spawner's x,z coordinates - not the spawner itself.
+-     * <br>
+-     * It is 2 blocks high, centered on the spawner's y-coordinate (its bottom);
+-     * thus allowing mobs to spawn as high as its top surface and as low
+-     * as 1 block below its bottom surface.
+-     * <br>
+-     * Default value is 4.
+-     *
+-     * @return the spawn range
+-     */
+-    public int getSpawnRange();
+-
+-    /**
+-     * Set the new spawn range.
+-     * <br>
+-     *
+-     * @param spawnRange the new spawn range
+-     * @see #getSpawnRange()
+-     */
+-    public void setSpawnRange(int spawnRange);
```

Comment:
I don't think this should be removed. I imagine the javadocs are the same, and this just creates more diff.

Reply:
Having created the PaperSpawnerHolder interface, I now have to remove these methods because there is a compile error otherwiese. For some reason the PaperSpawnerHolder interface cant implement methods from the CreatureSpawner interface, even though the methods have the same signature because it only provides default implementations for the methods of the SpawnerHolder interface. It is hard to explain, I hope you understand what I mean

Reply:
That doesn't make any sense. There shouldn't be a compile error if the method in the CreatureSpawner interface is just overriding one in the SpawnerHolder interface. In fact I just did that right now, and there was no compile error. You need to make sure that the methods you added to the SpawnerHolder interface exactly match the ones in the CreatureSpawner interface

Reply:
As you have suggested I have made sure that every signature from SpawnerHolder is exactly the same as in CreatureSpawnerHolder but it still gives me a compile error if I dont remove the overriding methods from CreatureSpawner.
Please try to compile this example, I think it will make it clear what I mean: https://gist.github.com/Lama06/b05e4f5c6226f3b7309a8a5348bd25d5

Reply:
Well that's cause you did the exact same thing with the methods in CraftCreatureSpawner. Don't delete the already implemented methods. Just like you shouldn't delete the methods in CreatureSpawner, to reduce diff, don't delete the methods in CraftCreatureSpawner.

Reply:
I have done what you have suggested and I now no longer delete the existing methods in CraftCreatureSpawner and CreatureSpawner.
Thanks for your help!